sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncAccountRepository.kt

```diff
@@ -322,6 +323,13 @@ class AppSyncAccountRepository @Inject constructor(
 
     override fun isSignedIn() = syncStore.isSignedIn()
 
+    override fun logoutAndJoinNewAccount(stringCode: String): Result<Boolean> {
+        logout(syncStore.deviceId.orEmpty()).takeIf { it is Error }?.let {
```

Comment:
this line is hard to understand at a glance. breaking it up into a few simpler lines would help readability

Reply:
```
        val thisDeviceId = syncStore.deviceId.orEmpty()
        return when (val result = logout(thisDeviceId)) {
            is Error -> result
            is Result.Success -> processCode(stringCode)
        }
```
That should be better.

Reply:
100x better, thanks!